scaleway/data_source_instance_image.go

```diff
@@ -0,0 +1,150 @@
+)
+
+func dataSourceScalewayInstanceImage() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceScalewayInstanceImageRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "exact name of the desired image",
+				ConflictsWith: []string{"image_id"},
+			},
+			"image_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "exact name of the desired image",
+				ConflictsWith: []string{"name", "architecture"},
+			},
+			"architecture": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Default:       instance.ArchX86_64.String(),
+				Description:   "architecture of the desired image",
+				ConflictsWith: []string{"image_id"},
+			},
+			"zone":            zoneSchema(),
+			"organization_id": organizationIDSchema(),
+
+			"public": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "indication if the image is public",
+			},
+			"bootscript_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the bootscript associated with this image",
+			},
+			"root_volume_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the root volume associated with this image",
+			},
+			"additional_volume_ids": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "The additional volume IDs attached to the image",
+			},
+			"from_server_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the server the image is originated from",
+			},
+			"creation_date": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "date when the image was created",
+			},
+			"modification_date": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "date when the image was updated",
+			},
+			"state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "state of the image",
+			},
+		},
+	}
+}
+
+func dataSourceScalewayInstanceImageRead(d *schema.ResourceData, m interface{}) error {
+	meta := m.(*Meta)
+	instanceApi, zone, err := getInstanceAPIWithZone(d, meta)
+	if err != nil {
+		return err
+	}
+
+	imageID, ok := d.GetOk("image_id")
+	if !ok {
+		res, err := instanceApi.ListImages(&instance.ListImagesRequest{
+			Zone: zone,
+			Name: expandStringPtr(d.Get("name")),
+			Arch: expandStringPtr(d.Get("architecture")),
+		})
+		if err != nil {
+			return err
+		}
+		if len(res.Images) == 0 {
+			return fmt.Errorf("no image found with the name %s and architecture %s", d.Get("name"), d.Get("architecture"))
+		}
+		if len(res.Images) > 1 {
+			return fmt.Errorf("%d images found with the same name %s and architecture %s", len(res.Images), d.Get("name"), d.Get("architecture"))
```

Comment:
Waht about the case we have 2 image names `ubuntu-bionic` and `ubuntu-bionic-ml` this condition will not work